staging/src/k8s.io/csi-translation-lib/plugins/portworx.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugins
+
+import (
+	"fmt"
+
+	"k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	PortworxVolumePluginName = "kubernetes.io/portworx-volume"
+	PortworxDriverName       = "pxd.portworx.com"
+)
+
+var _ InTreePlugin = &portworxCSITranslator{}
+
+type portworxCSITranslator struct{}
+
+func NewPortworxCSITranslator() InTreePlugin {
+	return &portworxCSITranslator{}
+}
+
+// TranslateInTreeStorageClassToCSI takes in-tree storage class used by in-tree plugin
+// and translates them to a storageclass consumable by CSI plugin
+func (p portworxCSITranslator) TranslateInTreeStorageClassToCSI(sc *storagev1.StorageClass) (*storagev1.StorageClass, error) {
+	if sc == nil {
+		return nil, fmt.Errorf("sc is nil")
+	}
+	sc.Provisioner = PortworxDriverName
```

Comment:
You may also need to translate the intree fstype key to the csi fstype key: https://github.com/kubernetes/kubernetes/blob/6e56b766b4d1393bfa95013cb5fa7564be47c9c9/staging/src/k8s.io/csi-translation-lib/plugins/gce_pd.go#L87

Reply:
In Portworx The actual flag is called `fs` and is the same across in-tree storage class and csi storage class

Reply:
Ok CSI has defined a first class fstype key that allows it to get passed down as a CSI parameter. You may want to consider updating the driver to support that, but this is fine for now since the driver doesn't support it.